### ISSUE TITLE
Can we force stop the script and restart from a checkpoint ?

### ISSUE THREAD

**Issue by abmitra84** [UKPLab/sentence-transformers#303]
As this is a single GPU computation so it is taking time for me to evaluate every 1000 against a large number of iterations. So I want to stop my script and change a few params and restart. Can I do it from the checkpoint in the output path mentioned below ? Just want to be sure I am not missing something as it looked doable. 

@nreimers 
```
train_batch_size = 16
triplet_reader = TripletReader('../datasets/wikipedia-sections-triplets', s1_col_idx=1, s2_col_idx=2, s3_col_idx=3, delimiter=',', quoting=csv.QUOTE_MINIMAL, has_header=True)
output_path = "output/training-wikipedia-sections-"+model_name+"-"+datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
num_epochs = 1
```

**Comment by nreimers** [comment]
Hi  @abmitra84 
Yes, you can stop the training and then continue.

But note some things:
- By default, only the model that performs best on the dev set is saved. This is not necessarily the model from the last epoch. But you could store the model by hand.

- The system uses a learning rate scheduler. By default, the learning rate (LR) is increased up to the number of steps you specify in the warm up phase. Then, the learning rate decreases up to 0 until the end of the training. If you continue training, the learning rate scheduler starts from scratch.

Best
Nils Reimers

**Comment by abmitra84** [comment]
Thanks for your response. 

About the saving of best model - I did notice it. I think its OK to start from the "last best" one. 

Good point on the learning_rate scheduler.
